src/client.cpp

```diff
@@ -240,57 +244,33 @@ PVR_ERROR GetChannels(ADDON_HANDLE handle, bool bRadio)
-    if (StreamUtils::ChannelSpecifiesInputstream(m_currentChannel))
-    {
-      // Channel has an inputstream class set so we only set the stream URL
-      StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
-    }
-    else
-    {
-      StreamType streamType = StreamUtils::GetStreamType(streamURL, m_currentChannel);
-      if (streamType == StreamType::OTHER_TYPE)
-        streamType = StreamUtils::InspectStreamType(streamURL);
-
-      // Using kodi's built in inputstreams
-      if (StreamUtils::UseKodiInputstreams(streamType))
-      {
-        std::string ffmpegStreamURL = StreamUtils::GetURLWithFFmpegReconnectOptions(streamURL, streamType, m_currentChannel);
-        StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, ffmpegStreamURL);
-
-        if (streamType == StreamType::HLS)
-          StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, PVR_STREAM_PROPERTY_INPUTSTREAMCLASS, PVR_STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG);
-      }
-      else // inputstream.adpative
-      {
-        StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
-        StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, PVR_STREAM_PROPERTY_INPUTSTREAMCLASS, "inputstream.adaptive");
-        StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, "inputstream.adaptive.manifest_type", StreamUtils::GetManifestType(streamType));
-        if (streamType == StreamType::HLS || streamType == StreamType::DASH)
-          StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, PVR_STREAM_PROPERTY_MIMETYPE, StreamUtils::GetMimeType(streamType));
-        if (streamType == StreamType::DASH)
-          StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, "inputstream.adaptive.manifest_update_parameter", "full");
-      }
-    }
 
-    Logger::Log(LogLevel::LEVEL_DEBUG, "%s - inputstream URL: %s", __FUNCTION__, streamURL.c_str());
+    std::string streamURL = m_currentChannel.GetStreamURL();
 
-    if (!m_currentChannel.GetProperties().empty())
-    {
-      for (auto& prop : m_currentChannel.GetProperties())
-        StreamUtils::SetStreamProperty(properties, iPropertiesCount, propertiesMax, prop.first, prop.second);
-    }
+    m_catchupController->ResetCatchupState(); // TODO: we need this currently until we have a way to know the stream stops.
+
+    // We always call the catchup controller regardless so it can cleanup state
+    // whether or not it supports catchup in case there is any houskeeping to do
+    std::map<std::string, std::string> catchupProperties;
+    m_catchupController->ProcessChannelForPlayback(m_currentChannel, catchupProperties);
+
+    std::string catchupUrl = m_catchupController->GetCatchupUrl(m_currentChannel);
```

Comment:
`const std::string` ? Or is it modified by `StreamUtils::SetAllStreamProperties`?

Reply:
Should be const, you're right.